presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1548,6 +1553,8 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
                 .collect(toImmutableMap(HiveColumnHandle::getName, column -> column.getHiveType().getType(typeManager)));
         Map<List<String>, ComputedStatistics> partitionComputedStatistics = createComputedStatisticsToPartitionMap(computedStatistics, partitionedBy, columnTypes);
 
+        Set<String> existingPartitions = getExistingPartitionNames(handle.getSchemaName(), handle.getTableName(), partitionUpdates);
```

Comment:
Why does this have to include partitions being appended to? Seems like we don't use this in the append path.
Are there cases where we will attempt to append and also overwrite/new?

Reply:
The write mode `APPEND` can only be set if the partition exist. The existence of partition is checked explicitly on the writer, so i thought that checking it here again doesn't make much sense.